### PR TITLE
HADOOP-16210. Update guava to 27.0-jre in hadoop-project trunk

### PR DESCRIPTION
**Pull request by bgaborg**
This is without the hadoop-tools modification (it will timeout on jenkins yetus after 5 hours, so I tested it manually.)

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 27 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 97 | Maven dependency ordering for branch |
| +1 | mvninstall | 1024 | trunk passed |
| +1 | compile | 981 | trunk passed |
| +1 | checkstyle | 194 | trunk passed |
| +1 | mvnsite | 328 | trunk passed |
| +1 | shadedclient | 1224 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-project |
| +1 | findbugs | 381 | trunk passed |
| +1 | javadoc | 211 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 22 | Maven dependency ordering for patch |
| +1 | mvninstall | 194 | the patch passed |
| +1 | compile | 908 | the patch passed |
| -1 | javac | 908 | root generated 15 new + 1481 unchanged - 1 fixed = 1496 total (was 1482) |
| +1 | checkstyle | 185 | root: The patch generated 0 new + 58 unchanged - 4 fixed = 58 total (was 62) |
| +1 | mvnsite | 285 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 604 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-project |
| -1 | findbugs | 102 | hadoop-common-project/hadoop-common generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| -1 | findbugs | 135 | hadoop-hdfs-project/hadoop-hdfs generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | javadoc | 199 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 18 | hadoop-project in the patch passed. |
| +1 | unit | 518 | hadoop-common in the patch passed. |
| -1 | unit | 6041 | hadoop-hdfs in the patch failed. |
| +1 | unit | 1019 | hadoop-hdfs-rbf in the patch passed. |
| +1 | unit | 341 | hadoop-mapreduce-client-core in the patch passed. |
| +1 | unit | 1125 | hadoop-yarn-services-core in the patch passed. |
| +1 | asflicense | 57 | The patch does not generate ASF License warnings. |
| | | 16162 | |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hadoop-common-project/hadoop-common |
|  |  Null passed for non-null parameter of writeXml(String, Writer) in org.apache.hadoop.conf.Configuration.writeXml(Writer)  At Configuration.java:of writeXml(String, Writer) in org.apache.hadoop.conf.Configuration.writeXml(Writer)  At Configuration.java:[line 3490] |
| FindBugs | module:hadoop-hdfs-project/hadoop-hdfs |
|  |  Null passed for non-null parameter of com.google.common.base.Preconditions.checkState(boolean, String, Object, Object, Object) in org.apache.hadoop.hdfs.qjournal.server.Journal.getPersistedPaxosData(long)  Method invoked at Journal.java:of com.google.common.base.Preconditions.checkState(boolean, String, Object, Object, Object) in org.apache.hadoop.hdfs.qjournal.server.Journal.getPersistedPaxosData(long)  Method invoked at Journal.java:[line 1064] |
| Failed junit tests | hadoop.hdfs.TestRollingUpgradeRollback |
|   | hadoop.hdfs.TestDistributedFileSystem |
|   | hadoop.hdfs.TestReadStripedFileWithDNFailure |
|   | hadoop.hdfs.TestReadStripedFileWithDecoding |
|   | hadoop.hdfs.TestReconstructStripedFileWithRandomECPolicy |
|   | hadoop.hdfs.server.datanode.TestBPOfferService |
|   | hadoop.hdfs.server.namenode.TestFsck |
|   | hadoop.hdfs.TestSafeMode |
|   | hadoop.hdfs.server.namenode.snapshot.TestFileContextSnapshot |
|   | hadoop.hdfs.TestWriteReadStripedFile |
|   | hadoop.hdfs.web.TestWebHdfsTimeouts |
|   | hadoop.hdfs.TestDFSStripedOutputStreamWithRandomECPolicy |
|   | hadoop.hdfs.TestFileAppend |
|   | hadoop.hdfs.TestReadStripedFileWithDecodingCorruptData |
|   | hadoop.hdfs.TestDFSInotifyEventInputStream |
|   | hadoop.hdfs.server.namenode.snapshot.TestRenameWithSnapshots |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-674/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/674 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  checkstyle  xml  |
| uname | Linux 1c8528e6a82a 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 0df9492 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/1/artifact/out/diff-compile-javac-root.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/1/artifact/out/new-findbugs-hadoop-common-project_hadoop-common.html |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/1/artifact/out/new-findbugs-hadoop-hdfs-project_hadoop-hdfs.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/1/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/1/testReport/ |
| Max. process+thread count | 4468 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-hdfs-project/hadoop-hdfs hadoop-hdfs-project/hadoop-hdfs-rbf hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-core U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-project/pom.xml [hunk 1]
```diff
@@ -92,7 +92,7 @@
     <spotbugs.version>3.1.0-RC1</spotbugs.version>
     <dnsjava.version>2.1.7</dnsjava.version>
 
-    <guava.version>11.0.2</guava.version>
+    <guava.version>27.0-jre</guava.version>
```
Comment (medb) [inline hadoop-project/pom.xml]: The latest available Guava version is `27.1-jre`. Is there a reason why `27.0-jre` was chosen instead?

Comment (bgaborg) [reply hadoop-project/pom.xml]: we decided to update a while ago with other components to 27.0-jre
it's easier to do the upgrade, 27.1-jre includes many other deprecations which would be way harder to do from 11.0.2 - so this is easier, precommit jenkins won't time out. If we really need to update 27.1-jre we can do that after fixing the new deprecations: https://issues.apache.org/jira/browse/HADOOP-16222

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 64 | Maven dependency ordering for branch |
| +1 | mvninstall | 1118 | trunk passed |
| +1 | compile | 993 | trunk passed |
| +1 | checkstyle | 215 | trunk passed |
| +1 | mvnsite | 312 | trunk passed |
| +1 | shadedclient | 1296 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-project |
| +1 | findbugs | 436 | trunk passed |
| +1 | javadoc | 236 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| +1 | mvninstall | 206 | the patch passed |
| +1 | compile | 1001 | the patch passed |
| -1 | javac | 1001 | root generated 15 new + 1481 unchanged - 1 fixed = 1496 total (was 1482) |
| +1 | checkstyle | 211 | root: The patch generated 0 new + 73 unchanged - 4 fixed = 73 total (was 77) |
| +1 | mvnsite | 314 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 4 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 709 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-project |
| +1 | findbugs | 477 | the patch passed |
| +1 | javadoc | 234 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 22 | hadoop-project in the patch passed. |
| +1 | unit | 534 | hadoop-common in the patch passed. |
| -1 | unit | 6067 | hadoop-hdfs in the patch failed. |
| +1 | unit | 1117 | hadoop-hdfs-rbf in the patch passed. |
| +1 | unit | 336 | hadoop-mapreduce-client-core in the patch passed. |
| +1 | unit | 1079 | hadoop-yarn-services-core in the patch passed. |
| +1 | asflicense | 52 | The patch does not generate ASF License warnings. |
| | | 16838 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.qjournal.client.TestQJMWithFaults |
|   | hadoop.hdfs.server.namenode.TestNameNodeMXBean |
|   | hadoop.hdfs.web.TestWebHdfsTimeouts |
|   | hadoop.hdfs.server.datanode.TestBPOfferService |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-674/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/674 |
| Optional Tests |  dupname  asflicense  xml  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  checkstyle  |
| uname | Linux 2d6891066658 4.4.0-138-generic #164~14.04.1-Ubuntu SMP Fri Oct 5 08:56:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / f96fb05 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/3/artifact/out/diff-compile-javac-root.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/3/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/3/testReport/ |
| Max. process+thread count | 2726 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-hdfs-project/hadoop-hdfs hadoop-hdfs-project/hadoop-hdfs-rbf hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-core U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-674/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by mackrorysd** [comment]
+1 - also did some more local testing and verified findbugs (although weirdly, I need to bump to findbugs 3.0.5 to not get errors, but they're unrelated to this patch). I've squashed the 2 commits in this PR and pushed.

**Comment by steveloughran** [comment]
Sean, there's a "close" button down below. If you don't see it, you may not be completely wired up ASF-permission wise

**Pull request closed without merging**